mgmtsystem/scvmm.py

```diff
@@ -271,13 +272,26 @@ def does_template_exist(self, template):
 
     def deploy_template(self, template, host_group, vm_name=None, **kwargs):
         timeout = kwargs.pop('timeout', 900)
+        if 'cpu' in kwargs and kwargs['cpu'] > self.provisioning['cpu']:
+            vm_cpu = kwargs['cpu']
+        else:
+            vm_cpu = self.provisioning['cpu']
+        if 'ram' in kwargs and kwargs['ram'] > self.provisioning['ram']:
+            vm_ram = kwargs['ram']
+        else:
+            vm_ram = self.provisioning['ram']
         script = """
         $tpl = Get-SCVMTemplate -Name "{template}" -VMMServer $scvmm_server
         $vm_host_group = Get-SCVMHostGroup -Name "{host_group}" -VMMServer $scvmm_server
         $vmc = New-SCVMConfiguration -VMTemplate $tpl -Name "{vm_name}" -VMHostGroup $vm_host_group
         Update-SCVMConfiguration -VMConfiguration $vmc
-        New-SCVirtualMachine -Name "{vm_name}" -VMConfiguration $vmc
-        """.format(template=template, vm_name=vm_name, host_group=host_group)
+        New-SCVirtualMachine -Name "{vm_name}" -VMConfiguration $vmc `
```

Comment:
Just so the process is clear, the VM is created exactly as defined in the template.  Then it is modified to fit the new cpu and ram settings.  That's why we can't make it smaller, only larger.  If I send the smaller values through via winrm, PowerShell will return a script execution error and abort the rest of the invoke-command contents.